lib/cell.rb

```diff
@@ -0,0 +1,46 @@
+require './lib/ship'
+
+class Cell
+  attr_reader :coordinate, :ship, :fire_upon
+
+  def initialize(coordinate)
+    @ship = nil
+    @coordinate = coordinate
+    @fired_upon = false
+    @render = "."
+  end
+
+  def empty?
+    if self.ship == nil
+      true
+    else
+      false
+    end
+  end
+
+  def place_ship(ship_type)
+    @ship = ship_type
+    self.ship
+  end
+
+  def fire_upon
+    @fired_upon = true
+    if self.ship != nil
+    @ship.health -= 1
+  end
```

Comment:
The ends got a little wonky here I think. The if statement needs and end (and the @ship.health -= 1 to be indented. You have an end floating below between your methods which is why it wasn't throwing an error I think. 

And then same thing with the self.ship here, this can be just @ship and it still passes!